### PR TITLE
[Storage] Make samples executable

### PR DESCRIPTION
**Pull request by rakshith91**
- Make samples executable
- Add read me to samples
- Fix dedents
- add missing rtypes

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/storage/azure-storage-blob/samples/blob_samples_common.py [hunk 1]
```diff
@@ -193,3 +213,11 @@ def start_copy_blob_from_url_and_abort_copy(self):
 
         finally:
             blob_service_client.delete_container("copyblobcontainer")
+
+if __name__ == '__main__':
+    sample = CommonBlobSamples()
+    sample.blob_snapshots()
+    sample.soft_delete_and_undelete_blob()
+    # sample.delete_multiple_blobs()
```
Comment (annatisch) [inline sdk/storage/azure-storage-blob/samples/blob_samples_common.py]: why did this one get skipped?

Comment (annatisch) [reply sdk/storage/azure-storage-blob/samples/blob_samples_common.py]: Discussed offline

File: sdk/storage/azure-storage-blob/samples/README.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+languages:
+  - python
+products:
+  - azure
+  - azure-storage-blob
+---
+
+# Azure Storage Blob client library for Python Samples
+
+These are code samples that show common scenario operations with the Azure Storage Blob client library.
```
[the remark this answers is not in the document]
Comment (rakshith91) [reply sdk/storage/azure-storage-blob/samples/README.md]: fixed

**Pull request merged**